cdisc_rules_engine/utilities/data_processor.py

```diff
@@ -41,196 +42,136 @@ def __init__(self, data_service=None, cache: CacheServiceInterface = None):
             data_service or DataServiceFactory(config, self.cache).get_data_service()
         )
 
-    @staticmethod
-    def calc_min(dataframe, target, grouping: List = None, **kwargs):
-        if not grouping:
-            return dataframe[target].min()
+    def min(self, params: OperationParams) -> Union[int, float, pd.DataFrame]:
```

Comment:
PR looks really good. My only  suggestion would be that we may want to do something like we did with the data services, and make operations a class by themselves and allow users to register operations. That way the language itself is extensible. It is quite a big change so I could see it being a new issue if we want to do it